src/groups/mqb/mqbc/mqbc_clusternodesession.cpp

```diff
@@ -82,11 +82,14 @@ void ClusterNodeSession::teardown()
     // executed by the *DISPATCHER* thread
 
     // Release all queue handles that were associated with this session
-    for (QueueHandleMapIter it = d_queueHandles.begin();
-         it != d_queueHandles.end();
-         ++it) {
-        mqbi::QueueHandle* handle = it->second.d_handle_p;
+    QueueHandleMapIter qit = d_queueHandles.begin();
+
+    while (qit != d_queueHandles.end()) {
+        mqbi::QueueHandle* handle = qit->second.d_handle_p;
+        BSLS_ASSERT_SAFE(handle);
+
         handle->drop();
+        d_queueHandles.erase(qit++);
```

Comment:
I think this variant is more clear. It matters because we are modifying a collection when we iterate over it
```suggestion
        qit = d_queueHandles.erase(qit);
```